### DISCUSSION TITLE
Is there a way of using partial masking with an inpainting vae?

### DISCUSSION THREAD

**Discussion by Ferniclestix** [comfyanonymous/ComfyUI#1122] [Q&A]
The inpainting vae seems very basic, it completley wipes the masked area of information based on the mask and then selectively inpaints this area.
This means the inpainting is often going to be significantly compromized as it has nothing to go off and uses none of the original image as a clue for generating an adjusted area. lowering the denoising settings simply shifts the output towards the neutral grey that replaces the masked area.
I feel like there should be an option where i can use some of the original image for generating during inpainting like alot of other interfaces are able to do.

And I guess im wondering if anyone has a work around. sick of being forced to do whole image inpainting when other interfaces do not suffer from this limitation.

Edit- AH, found it... I think? 'set latent noise mask' seeems to be doing as asked. more experiments needed.

**Comment by ltdrdata** [comment]
> The inpainting vae seems very basic, it completley wipes the masked area of information based on the mask and then selectively inpaints this area. This means the inpainting is often going to be significantly compromized as it has nothing to go off and uses none of the original image as a clue for generating an adjusted area. lowering the denoising settings simply shifts the output towards the neutral grey that replaces the masked area. I feel like there should be an option where i can use some of the original image for generating during inpainting like alot of other interfaces are able to do.
> 
> And I guess im wondering if anyone has a work around. sick of being forced to do whole image inpainting when other interfaces do not suffer from this limitation.
> 
> Edit- AH, found it... I think? 'set latent noise mask' seeems to be doing as asked. more experiments needed.

It seems like you should take a look at the 'detailer' in the Impact Pack. While 'Set Latent Noise Mask' updates only the masked area, it takes a long time to process large images because it considers the entire image area. In the Impact Pack, there's a technique that involves cropping the area around the mask by a certain size, processing it, and then recompositing it.

https://github.com/ltdrdata/ComfyUI-Impact-Pack

**Reply by Ferniclestix** [reply]
oh thats cool, ill take a look, thanks :D  love your nodes and stuff btw
